src/Libraries/RevitNodes/Elements/DirectShape.cs

```diff
@@ -0,0 +1,471 @@
+            }
+
+            if (geometry is Autodesk.DesignScript.Geometry.Solid || geometry is Autodesk.DesignScript.Geometry.Surface)
+            {
+                return new DirectShape(geometry, name, new ElementId(category.Id), new ElementId(material.Id));
+            }
+
+            throw new ArgumentException(Revit.Properties.Resources.DirectShapeInvalidArgument);
+        }
+
+        /// <summary>
+        /// Create a Revit DirectShape given some mesh, a name for the shape, category, and material.
+        /// The category of a DirectShape cannot be changed after placing a DirectShape, so
+        /// a new DirectShape will be generated if the category input is changed.
+        /// </summary>
+        /// <param name="mesh">a mesh that will be placed in the Revit model as a DirectShape</param>
+        /// <param name="name">a string name for the directshape</param>
+        /// <param name="category">must be a top level built-in category</param>
+        /// <param name="material">a material to apply to the faces of the DirectShape</param>
+        /// <returns>a DirectShape Element</returns>
+        public static DirectShape ByMesh(Autodesk.DesignScript.Geometry.Mesh mesh,
+            Category category,
+            [DefaultArgumentAttribute(" DirectShape.DynamoPreviewMaterial")]Material material,
+           string name = DEFAULT_NAME)
+        {
+            if (mesh == null)
+            {
+                throw new ArgumentNullException("mesh");
+            }
+
+            if (name == null)
+            {
+                throw new ArgumentNullException("name");
+            }
+
+            if (category == null)
+            {
+                throw new ArgumentNullException("category");
+            }
+
+            if (material == null)
+            {
+                throw new ArgumentNullException("material");
+            }
+
+            return new DirectShape(mesh, name, new ElementId(category.Id), new ElementId(material.Id));
+        }
+
+        #endregion
+
+        #region static constructors
+
+        /// <summary>
+        /// Construct a DirectShape from the Revit document. 
+        /// </summary>
+        /// <param name="directShape"></param>
+        /// <param name="isRevitOwned"></param>
+        /// <returns></returns>
+        internal static DirectShape FromExisting(Autodesk.Revit.DB.DirectShape directShape, bool isRevitOwned)
+        {
+            return new DirectShape(directShape)
+            {
+                IsRevitOwned = isRevitOwned
+            };
+        }
+
+        static DirectShape()
+        {
+            DynamoPreviewMaterial = Material.ByName(Document.GetElement(MaterialsManager.Instance.DynamoMaterialId).Name);
+            DYNAMO_DIRECTSHAPE_APP_GUID = Guid.Parse("03deaabe-6989-4d20-ba68-557001d2f45c");
+        }
+
+        #endregion
+
+        public override string ToString()
+        { 
+            return InternalDirectShape.Name;
```

Comment:
Use the style we use with the DisplayObject:

https://github.com/DynamoDS/Dynamo/blob/master/src/Libraries/Display/Display.cs#L144